File: University_Documentations/LearnListView/app/src/main/java/com/example/learnlistview/MainActivity.java
package com.example.learnlistview;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ListView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    ListView listView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        listView = (ListView) findViewById(R.id.listViewCountries);

        String[] countries = getResources().getStringArray(R.array.countries);


        ArrayAdapter<String> countriesAdapter = new ArrayAdapter<>(this,android.R.layout.simple_expandable_list_item_1,countries);
        listView.setAdapter(countriesAdapter);

        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                Toast.makeText(getApplicationContext(),countries[position],Toast.LENGTH_SHORT).show();
            }
        });
    }
}

File: CookingRecipeApp/app/src/main/java/com/example/cookingrecipeapp/MainActivity.kt
package com.example.cookingrecipeapp

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import androidx.navigation.Navigation

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        val menuInflater = menuInflater
        menuInflater.inflate(R.menu.add_recipe,menu)

        return super.onCreateOptionsMenu(menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if(item.itemId == R.id.addRecipeItem){
            val action = ListFragmentDirections.actionListFragmentToRecipeFragment()
            action.locationBeforeRecipeFragment = LocationBeforeRecipeFragment.FROM_NEW
            Navigation.findNavController(this@MainActivity,R.id.fragmentContainerView).navigate(R.id.action_listFragment_to_recipeFragment)
        }

        return super.onOptionsItemSelected(item)
    }
}